### ISSUE TITLE
[BUG] [iOS] [AzureCommunicationCommon] -Wdocumentation warning in CommunicationTokenCredential.swift

### ISSUE THREAD

**Issue by ryanespiritu** [Azure/azure-sdk-for-ios#1038]
**Describe the bug**
Seeing a clang warning (-Wdocumentation) for the incorrectly named `option` parameter in `CommunicationTokenCredential.swift`.

***Exception or Stack Trace***
```
.../AzureCommunicationCommon/AzureCommunicationCommon/AzureCommunicationCommon.framework/Headers/AzureCommunicationCommon-Swift.h:261:12: error: parameter 'option' not found in the function declaration [-Werror,-Wdocumentation]
/// \param option Options for how the token will be refreshed
           ^~~~~~
.../AzureCommunicationCommon/AzureCommunicationCommon/AzureCommunicationCommon.framework/Headers/AzureCommunicationCommon-Swift.h:261:12: note: did you mean 'options'?
/// \param option Options for how the token will be refreshed
           ^~~~~~
           options
```

**To Reproduce**
1. Pull version `1.0.2` of `AzureCommunicationCommon` into Xcode project
2. Build Xcode project
3. Observe warning

***Code Snippet***
```
/**
 Creates a CommunicationTokenCredential that automatically refreshes the token.
 - Parameters:
 - option: Options for how the token will be refreshed
 - Throws: `NSError` if the provided token is not a valid user token. `userInfo` contains `message` key for reason.
*/
public init(withOptions options: CommunicationTokenRefreshOptions) throws {
   ...
}
```

**Expected behavior**
`option` parameter should be properly referred to as `options` in the doc comment to fix warning.

**Setup (please complete the following information):**
 - OS: iOS
 - IDE: Xcode
 - AzureCommunicationCommon 1.0.2

**Comment by tjprescott** [comment]
This is indeed a bug in the docstring. Thanks for bringing it to our attention @ryanespiritu.

**Comment by JoshuaLai** [comment]
I have created a quick PR for this fix. Thanks!

**Comment by tjprescott** [comment]
Fixed in #1049.
